src/sickle.ts

```diff
@@ -66,6 +77,40 @@ class Annotator {
           this.writeNode(node);
         }
         break;
+      case ts.SyntaxKind.InterfaceDeclaration:
+        this.writeRange(node.getFullStart(), node.getEnd());
+        if (node.flags & ts.NodeFlags.Ambient) {
+          // It's a "declare interface ..."; write a Closure interface
+          // declaration so we can @export all properties.
+          let decl = <ts.InterfaceDeclaration>node;
+          let className = decl.name.getText();
+          let props: TypedProperty[] = [];
+          for (let member of decl.members) {
+            // Members can include things like index signatures, for e.g.
+            //   interface Foo { [key: string]: number; }
+            // For now, just die unless all the members are regular old
+            // properties.
+            if (member.kind != ts.SyntaxKind.PropertySignature) {
+              this.fail(`unhandled member of kind ${ts.SyntaxKind[member.kind]}`);
+            }
+            let prop = <ts.PropertySignature>member;
+            props.push({
+              name: prop.name.getText(),
+              type: prop.type,
+              extraTags: '@export',
+            });
+          }
+          // Note: be careful with this code!  It's easy to make something
+          // that Closure will accept but then also have it silently rename
+          // fields you intended to @export.
+          // @record tells Closure that this type is structural (not sure
+          // how well that works), while @struct makes it complain if you
+          // accidentally add a field that isn't declared.
+          this.emit('\n/** @record @struct */\n');
```

Comment:
Would it make sense to insert a comment into the output, explaining to the baffled debugger what this is doing here?